numpy/lib/_function_base_impl.py

```diff
@@ -2600,16 +2602,14 @@ def _vectorize_call(self, func, args):
         else:
             ufunc, otypes = self._get_ufunc_and_otypes(func=func, args=args)
 
-            # Convert args to object arrays first
-            inputs = [asanyarray(a, dtype=object) for a in args]
-
-            outputs = ufunc(*inputs)
+            inputs = [asanyarray(a) for a in args]
```

Comment:
You should be able to just delete this, I think.

Reply:
These inputs will be very different for the `ufunc`.

Reply:
That should be OK ufuncs basically call `asanyarray` first thing. The one subtlety would be the `dtype=object`, but if you remove it anyway then I don't think there should be a problem.

(I have to think about whether this is OK or not, converting to `dtype=object` ahead of time may actually be slightly beneficial if the input isn't an array already at least.)

Reply:
I think the `dtype=object` should not be removed, because `vectorize(f)([np.array(1)])` is not the same as `vectorize(f)(np.array([1]))`.